### PR TITLE
Add draft release notes for 0.0.123

### PR DESCRIPTION
**Pull request by TheBlueMatt**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,95 @@
...
+ * After initiating an outbound channel, a peer disconnection no longer results
+   in immediate channel closure. Rather, if the peer is reconnected before the
+   channel times out LDK will automatically retry opening it (#2725).
+ * `PaymentPurpose` now has separate variants for BOLT12 payments, which
+   include fields from the `invoice_request` as well as the `OfferId` (#2970).
+ * `ChannelDetails` now includes a list of in-flight HTLCs (#2442).
+ * `Event::PaymentForwarded` now includes `skimmed_fee_msat` (#2858).
+ * The `hashbrown` dependency has been upgraded and the use of `ahash` as the
+   no-std hash table hash function has been removed. As a consequence, LDK's
+   `Hash{Map,Set}`s no longer feature several constructors when LDK is build
```
Comment (jkczyz) [inline CHANGELOG.md:20]: s/build/built

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,95 @@
...
+   with no-std, see the `util::hash_tables` module instead. On platforms which
```
Comment (jkczyz) [inline CHANGELOG.md:21]: s/,/;

Comment (jkczyz) [inline CHANGELOG.md:21]: /s/which/that

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,95 @@
...
+ * `NodeId::from_slice` is now provided (#2942).
+ * `ChannelManager` deserialization may now fail with `DangerousValue` when
+    LDK's persistence API was violated (#2974).
+
+## Bug Fixes
+ * `Future`s used as an `std::...::Future` could grow in size unbounded if it
+   was never woken. For those not using async persistence and using the async
+   `lightning-background-processor`, this could cause a memory leak in the
+   `ChainMonitor` (#2894).
+ * Inbound channel requests which fail in
```
Comment (jkczyz) [inline CHANGELOG.md:62]: s/which/that

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,95 @@
...
+   `ChannelManager::accept_inbound_channel` would previously have stalled from
+   the peer's perspective as no `error` message was sent (#2953).
+ * Blinded path construction has been tuned to select paths more likely to
+   succeed, improving BOLT12 payment reliability (#2911, #2912, XXX?).
```
Comment (jkczyz) [inline CHANGELOG.md:66]: Is another PR expected here?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,95 @@
...
+ * After a reorg, `lightning-transaction-sync` could have failed to follow a
+   transaction which LDK needed information about (#2946).
```
Comment (jkczyz) [inline CHANGELOG.md:68]: s/which/that

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,95 @@
...
+   establish a connection (#2993).
+ * `ChannelUpdate` messages broadcasted for our own channel closures are now
+   slightly more robust (#2731).
+ * In exceedingly rare cases (no cases of this are known), LDK may have created
+   an invalid serialization for a `ChannelManager` (#2998).
+ * Message processing latency handling BOLT12 payments has been reduced (#2881).
+
+## Node Compatibility
+ * LDK's blinded paths were inconsistent with other implementations in several
+   ways, which have been addressed (#2856, #2936, #2945, XXX)
```
Comment (valentinewallace) [inline CHANGELOG.md:81]: I believe the XXX can be removed.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,112 @@
...
+ * Bounds in `lightning-background-processor` were simplified by using
+   `AChannelManager` (#2963).
+ * The `Persist` impl for `KVStore` no longer requires `Sized`, allowing for
+   the use of `dyn KVStore` as `Persist` (#2883, #2976).
+ * `From<PaymentPreimage>` is now implemented for `PaymentHash` (#2918).
+ * `NodeId::from_slice` is now provided (#2942).
+ * `ChannelManager` deserialization may now fail with `DangerousValue` when
+    LDK's persistence API was violated (#2974).
+
+## Bug Fixes
```
Comment (valentinewallace) [inline CHANGELOG.md:57]: #3021, #2971, #2951 could be mentioned?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,143 @@
...
+   by default does not set it for those not using a `P2PGossipSync` (e.g. those
+   using RGS). This change was reverted in CLN v24.02.2 however for now LDK
+   always sets the `gossip_queries` feature. This change is expected to be
+   reverted in a future LDK release (#2959).
+
+## Security
+0.0.123 fixes a denial-of-service vulnerability which we believe to be reachable
+from untrusted input when parsing invalid BOLT11 invoices containing non-ASCII
+characters.
+ * BOLT11 invoices with non-ASCII characters in the human-readable-part my cause
```
Comment (valentinewallace) [inline CHANGELOG.md:115]: nit: may cause

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,143 @@
...
+   an out-of-bounds read attempt leading to a panic (#3054). Note that all
+   BOLT11 invoices containing non-ASCII characters are invalid.
+
+In total, this release features 150 files changed, 19307 insertions, 6306
+deletions in 360 commits since 0.0.121 from 17 authors, in alphabetical order:
```
Comment (valentinewallace) [inline CHANGELOG.md:120]: s/121/122?

**Review by valentinewallace — APPROVED** [review]

**Review by jkczyz — APPROVED** [review]

**Pull request merged**